egs/wsj/s5/utils/queue.pl

```diff
@@ -488,11 +538,11 @@ ()
         # Don't run qstat too often, avoid stress on SGE; the if-condition above
         # is designed to check every 10 waits at first, and eventually every 50
         # waits.
-        if ( -f $f ) { next; }  #syncfile appeared: OK.
+        if ( -f $f ) { check_status; next; }  #syncfile appeared: OK.
         my $output = `qstat -j $sge_job_id 2>&1`;
         my $ret = $?;
-        if ($ret >> 8 == 1 && $output !~ m/qmaster/ &&
-            $output !~ m/gdi request/) {
+        if (( $ret >> 8 == 1 && $output !~ m/qmaster/ &&
+            $output !~ m/gdi request/) || ($ret == 0) ) {
```

Comment:
not sure what the `|| ($ret == 0)` is for; this if-statement seems to be for the case when the job does not exist.  If you're going to change things this much, it needs to be explained clearly at the very least..